### ISSUE TITLE
FALSE POSITIVE: `OS_OPEN_STREAM` reports after assigning `PrintWriter.append()` back to the same variable

### ISSUE THREAD

**Issue by lynae99** [spotbugs/spotbugs#4274]
Hi, I found a false positive in SpotBugs 4.10.3 when the result of `PrintWriter.append()` is assigned back to the same local variable before the writer is closed in `finally`.

**Affected tool**

`SpotBugs 4.10.3`

**Affected checker**

`SpotBugs OS_OPEN_STREAM`

**Minimal reproducer**

```java
import java.io.IOException;
import java.io.PrintWriter;

class SpotBugsPrintWriterAppend {
  void test() throws IOException {
    PrintWriter writer = null;
    try {
      writer = new PrintWriter("file.txt");
      writer = writer.append(null);
    } finally {
      if (writer != null) writer.close();
    }
  }
}
```

**Reproduction command**

```bash
spotbugs -version
javac -d classes spotbugs-os-open-stream-fp-printwriter-append.java
spotbugs -textui -effort:max -low -visitors FindOpenStream classes
```

**Current behavior**

SpotBugs reports `OS_OPEN_STREAM` at line 8. The report disappears if `writer.append(null)` is called without assigning its return value back to `writer`.

**Expected behavior**

SpotBugs should not report `OS_OPEN_STREAM` because `PrintWriter.append()` returns the same writer, and that writer is closed in `finally`.

**Comment by angus-guo** [comment]
I'll work on this:

- Preserve the receiver's resource identity for the three `PrintWriter.append` overloads in `StreamFrameModelingVisitor`.
- Add regression cases for reassignment/chaining followed by closing, and verify that unclosed writers are still reported.
- Keep other method calls and existing resource-escape handling unchanged.
